### PR TITLE
Add `paddle.nn.loss.MSELoss`

### PR DESCRIPTION
**Pull request by willthefrog**
test=develop

Add new public API `paddle.nn.loss.MSELoss(reduction='sum'|'mean'|'none')`.

Usage example:
```python
import numpy as np
from paddle import fluid
import paddle.fluid.dygraph as dg
from paddle.nn.loss import MSELoss

mse_loss = MSELoss()
input = fluid.data(name="input", shape=[1])
label = fluid.data(name="label", shape=[1])
place = fluid.CPUPlace()
input_data = np.array([1.5]).astype("float32")
label_data = np.array([1.7]).astype("float32")

# declarative mode
output = mse_loss(input,label)
exe = fluid.Executor(place)
exe.run(fluid.default_startup_program())
output_data = exe.run(
    fluid.default_main_program(),
    feed={"input":input_data, "label":label_data},
    fetch_list=[output],
    return_numpy=True)
print(output_data)
# [array([0.04000002], dtype=float32)]

# imperative mode
with dg.guard(place) as g:
    input = dg.to_variable(input_data)
    label = dg.to_variable(label_data)
    output = mse_loss(input, label)
    print(output.numpy())
    # [0.04000002]
```

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_mse_loss.py [hunk 1]
```diff
@@ -47,5 +47,115 @@ def test_mse_loss(self):
...
+    def test_NNMseLoss_mean(self):
+        input_np = np.random.uniform(0.1, 0.5, (2, 3)).astype("float32")
+        label_np = np.random.uniform(0.1, 0.5, (2, 3)).astype("float32")
+        prog = fluid.Program()
+        startup_prog = fluid.Program()
+        place = fluid.CUDAPlace(0) if fluid.core.is_compiled_with_cuda(
+        ) else fluid.CPUPlace()
+        with fluid.program_guard(prog, startup_prog):
+            input = fluid.layers.data(
+                name='input', shape=[2, 3], dtype='float32')
```
Comment (heavengate) [inline python/paddle/fluid/tests/unittests/test_mse_loss.py]: 元素个数需要>100

File: python/paddle/fluid/tests/unittests/test_mse_loss.py [hunk 2]
```diff
@@ -47,5 +47,115 @@ def test_mse_loss(self):
...
+            dy_result = dy_ret.numpy()
+
+        sub = input_np - label_np
+        expected = (sub * sub)
+        self.assertTrue(np.allclose(static_result, expected))
+        self.assertTrue(np.allclose(static_result, dy_result))
+        self.assertTrue(np.allclose(dy_result, expected))
+        self.assertTrue(dy_result.shape, [1])
+
+
```
Comment (heavengate) [inline python/paddle/fluid/tests/unittests/test_mse_loss.py:179]: 建议再补充下3-D 4-D 输入的单测

File: python/paddle/fluid/tests/unittests/test_mse_loss.py (hunk 1 above)
Comment (willthefrog) [reply python/paddle/fluid/tests/unittests/test_mse_loss.py]: done

File: python/paddle/fluid/tests/unittests/test_mse_loss.py (hunk 2 above)
Comment (willthefrog) [reply python/paddle/fluid/tests/unittests/test_mse_loss.py:179]: done

**Review by heavengate — DISMISSED** [review]
LGTM

**Review by heavengate — DISMISSED** [review]

**Review by heavengate — DISMISSED** [review]

**Review by heavengate — APPROVED** [review]
